metals/src/main/scala/scala/meta/internal/metals/InteractiveSemanticdbs.scala

```diff
@@ -160,6 +161,17 @@ final class InteractiveSemanticdbs(
   }
 
   private def compile(source: AbsolutePath, text: String): s.TextDocument = {
+    if (source.isJavaFilename)
+      javaInteractiveSemanticdb.fold(s.TextDocument())(
+        _.textDocument(source, text)
+      )
```

Comment:
Do you have a strong preference for `fold`?
```suggestion
      javaInteractiveSemanticdb.map(_.textDocument(source, text)).getOrElse(s.TextDocument())
```

Reply:
No, but what do you have against of it? :detective:

Reply:
I just prefer a more explicit way - pattern matching, `gerOrElse`. Fold seems a little bit obscure

Reply:
@kpodsiad Don't you mind if I'll ignore this? :smiling_face_with_tear: 

Fold is a legal function in our codebase.
For me `fold` here is a kind of signal that it's a bit different from usual plain data operation.